omop_file_validator.py

```diff
@@ -92,6 +100,31 @@ def cast_type(cdm_column_type, value):
         return value
 
 
+def date_format_valid(ptn, date_str, fmt='%Y-%m-%d'):
+    """Check if a date string matches a certain pattern and is compilable into a datetime object
+
+    :param ptn: A regex pattern
+    :type ptn: string
+    :param date_str: 
+    :type date_str: string
+    :param fmt: A C standard-compliant date format, defaults to '%Y-%m-%d'
+    :type fmt: str, optional
+    :return: A boolean indicating if date string matches the regex pattern
+    :rtype: bool
+    """
+
+    if not re.match(ptn, date_str):
+        return False
+
+    try:
```

Comment:
I tested this using an out of range date and and get the standard `"Invalid date format. Expecting ""YYYY-MM-DD"` and `"Invalid timestamp format. Expecting ""YYYY-MM-DD hh:mm:ss""` 

Do you think this will make the sites confused because on quick glance, they'll see that it's in the proper format but an invalid date? Not sure if adding an error message stating that an invalid date was inputted to clear any confusion. I will ask the other engineers.

Reply:
The team believes that because the check as it is now is adherence to the data model and file format so no constraining of the date is necessary. In the future, a check to make sure the date is in a valid range could be added after the file check logic is moved to the curation repository.

Reply:
Thanks for the review, Danyel! I see that you approved the request, but are you saying that an out of range date like '2020-02-30' should be allowed for now?

Reply:
No, they're already not allowed because of these lines and the regex so it's working fine. Dates like this will raise an error however it's not a "date out of range error" it raises an "invalid date format" or "invalid date time format" error which could possibly cause confusion to the sites. However for now, we're leaving this as is since the sites will know that something is up with the date, that's good enough for now.

Hope that clears up your confusion 😅

Reply:
Gotcha, thanks. :)